entity.go

```diff
@@ -182,6 +206,20 @@ func (id *IndexDefinition) Clone() *IndexDefinition {
 	}
 }
 
+func (id *IndexDefinition) String() string {
+	return fmt.Sprintf("%+v", *id)
+}
+
+func (id *IndexDefinition) equal(other *IndexDefinition) error {
```

Comment:
so instead of using `DeepEqual`, we will our own equal, is it risky which might introduce unexpected unequal thing.
I don't get it what's reason we are giving up DeepEqual. If we want more error info from `DeepEqual`, just use `%+v` rather than `%v` is good enough, that's how I debug before...

Reply:
I don't understand this. What does DeepEqual buy us?

Reply:
To elaborate further: the reason we should not use DeepEqual here is because we don't necessarily want to compare each member of a struct to check equality. 

For instance, an object may have an internal unique ID field, or some pre-compiled cached stuff, or ... We want to make sure the user did not screw up when the schema change was made, so the code should only check the things that came from the user. We should not check fields whose values may change between executions of the program.

Reply:
I am convinced and understand more that DeepEqual has done too much. Thanks for the detailed explanation.